### ISSUE TITLE
Importing course fails

### ISSUE THREAD

**Issue by natea** [regisb/openedx-docker#7]
When I try to run the command `make import-demo-course`

I get some errors and the course doesn't appear to have imported correctly because I get a 404 error when I click on the course, even though it appears in the catalog:

```
2018-02-17 00:07:48,144 INFO 20 [dd.dogapi] dog_stats_api.py:66 - Initializing dog api to use statsd: localhost, 8125
Importing.  Data_dir=../data, source_dirs=('../edx-demo-course',)
2018-02-17 00:07:48,389 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:48,405 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:48,415 ERROR 20 [xmodule.vertical_block] vertical_block.py:131 - Unable to load child when parsing Vertical. Continuing...
Traceback (most recent call last):
  File "/openedx/edx-platform/common/lib/xmodule/xmodule/vertical_block.py", line 128, in definition_from_xml
    child_block = system.process_xml(etree.tostring(child, encoding='unicode'))
  File "/openedx/edx-platform/common/lib/xmodule/xmodule/modulestore/xml.py", line 196, in process_xml
    id_manager,
  File "/openedx/edx-platform/common/lib/xmodule/xmodule/x_module.py", line 1601, in xblock_from_node
    self._convert_reference_fields_to_keys(block)  # difference from XBlock.runtime
  File "/openedx/edx-platform/common/lib/xmodule/xmodule/x_module.py", line 1645, in _convert_reference_fields_to_keys
    field_value = getattr(xblock, field.name)
  File "/usr/local/lib/python2.7/dist-packages/xblock/fields.py", line 513, in __get__
    value = self.from_json(field_data.get(xblock, self.name))
  File "/usr/local/lib/python2.7/dist-packages/xblock/fields.py", line 819, in from_json
    raise TypeError('Value stored in a List must be None or a list, found %s' % type(value))
TypeError: Value stored in a List must be None or a list, found <type 'str'>
2018-02-17 00:07:48,417 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:48,430 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:48,446 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:48,473 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:48,487 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:48,497 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:48,513 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:48,524 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:48,535 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:48,548 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:48,559 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:48,745 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:48,757 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:48,773 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:48,814 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:48,836 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:48,861 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:48,876 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:48,949 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:48,973 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:49,006 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:49,032 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:49,043 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:49,055 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:49,071 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:49,083 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:49,096 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:49,106 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:49,471 INFO 20 [openedx.core.djangoapps.content.course_overviews.models] models.py:150 - Creating course overview for course-v1:edX+DemoX+Demo_Course.
2018-02-17 00:07:49,544 INFO 20 [xmodule.modulestore.django] django.py:207 - Sent course_published signal to <function listen_for_course_publish at 0x7f9c67a3f410> with kwargs {'course_key': CourseLocator('edX', 'DemoX', 'Demo_Course', None, None)}. Response was: None
2018-02-17 00:07:49,545 INFO 20 [xmodule.modulestore.django] django.py:207 - Sent course_published signal to <function listen_for_course_publish at 0x7f9c4c9adaa0> with kwargs {'course_key': CourseLocator('edX', 'DemoX', 'Demo_Course', None, None)}. Response was: None
2018-02-17 00:07:49,546 INFO 20 [xmodule.modulestore.django] django.py:207 - Sent course_published signal to <function _update_block_structure_on_course_publish at 0x7f9c4c9b36e0> with kwargs {'course_key': CourseLocator('edX', 'DemoX', 'Demo_Course', None, None)}. Response was: None
2018-02-17 00:07:49,547 INFO 20 [xmodule.modulestore.django] django.py:207 - Sent course_published signal to <function _listen_for_course_publish at 0x7f9c4c7fd578> with kwargs {'course_key': CourseLocator('edX', 'DemoX', 'Demo_Course', None, None)}. Response was: None
2018-02-17 00:07:49,548 INFO 20 [xmodule.modulestore.django] django.py:207 - Sent course_published signal to <function _listen_for_course_publish at 0x7f9c4c7fda28> with kwargs {'course_key': CourseLocator('edX', 'DemoX', 'Demo_Course', None, None)}. Response was: None
2018-02-17 00:07:49,550 INFO 20 [xmodule.modulestore.django] django.py:207 - Sent course_published signal to <function trigger_update_xblocks_cache_task at 0x7f9c4c7fdb18> with kwargs {'course_key': CourseLocator('edX', 'DemoX', 'Demo_Course', None, None)}. Response was: None
2018-02-17 00:07:49,551 INFO 20 [xmodule.modulestore.django] django.py:207 - Sent course_published signal to <function _listen_for_course_publish at 0x7f9c4c7fd8c0> with kwargs {'course_key': CourseLocator('edX', 'DemoX', 'Demo_Course', None, None)}. Response was: None
2018-02-17 00:07:50,992 ERROR 20 [root] content.py:461 - Failed to generate thumbnail for asset-v1:edX+DemoX+Demo_Course+type@asset+block@jsmol_Al2O3.png. Exception: cannot identify image file <StringIO.StringIO instance at 0x7f9c4b4bcea8>
Traceback (most recent call last):
  File "/openedx/edx-platform/common/lib/xmodule/xmodule/contentstore/content.py", line 440, in generate_thumbnail
    with Image.open(source) as image:
  File "/usr/local/lib/python2.7/dist-packages/PIL/Image.py", line 2319, in open
    % (filename if filename else fp))
IOError: cannot identify image file <StringIO.StringIO instance at 0x7f9c4b4bcea8>
2018-02-17 00:07:51,092 ERROR 20 [root] content.py:461 - Failed to generate thumbnail for asset-v1:edX+DemoX+Demo_Course+type@asset+block@jsmol_1hxw.png. Exception: cannot identify image file <StringIO.StringIO instance at 0x7f9c4b4bf4d0>
Traceback (most recent call last):
  File "/openedx/edx-platform/common/lib/xmodule/xmodule/contentstore/content.py", line 440, in generate_thumbnail
    with Image.open(source) as image:
  File "/usr/local/lib/python2.7/dist-packages/PIL/Image.py", line 2319, in open
    % (filename if filename else fp))
IOError: cannot identify image file <StringIO.StringIO instance at 0x7f9c4b4bf4d0>
2018-02-17 00:07:54,224 INFO 20 [root] xml_importer.py:290 - No assets.xml file is present with asset metadata.
2018-02-17 00:07:56,144 INFO 20 [openedx.core.djangoapps.content.course_overviews.models] models.py:147 - Updating course overview for course-v1:edX+DemoX+Demo_Course.
2018-02-17 00:07:56,171 INFO 20 [xmodule.modulestore.django] django.py:207 - Sent course_published signal to <function listen_for_course_publish at 0x7f9c67a3f410> with kwargs {'course_key': CourseLocator('edX', 'DemoX', 'Demo_Course', None, None)}. Response was: None
2018-02-17 00:07:56,172 INFO 20 [xmodule.modulestore.django] django.py:207 - Sent course_published signal to <function listen_for_course_publish at 0x7f9c4c9adaa0> with kwargs {'course_key': CourseLocator('edX', 'DemoX', 'Demo_Course', None, None)}. Response was: None
2018-02-17 00:07:56,173 INFO 20 [xmodule.modulestore.django] django.py:207 - Sent course_published signal to <function _update_block_structure_on_course_publish at 0x7f9c4c9b36e0> with kwargs {'course_key': CourseLocator('edX', 'DemoX', 'Demo_Course', None, None)}. Response was: None
2018-02-17 00:07:56,174 INFO 20 [xmodule.modulestore.django] django.py:207 - Sent course_published signal to <function _listen_for_course_publish at 0x7f9c4c7fd578> with kwargs {'course_key': CourseLocator('edX', 'DemoX', 'Demo_Course', None, None)}. Response was: None
2018-02-17 00:07:56,174 INFO 20 [xmodule.modulestore.django] django.py:207 - Sent course_published signal to <function _listen_for_course_publish at 0x7f9c4c7fda28> with kwargs {'course_key': CourseLocator('edX', 'DemoX', 'Demo_Course', None, None)}. Response was: None
2018-02-17 00:07:56,175 INFO 20 [xmodule.modulestore.django] django.py:207 - Sent course_published signal to <function trigger_update_xblocks_cache_task at 0x7f9c4c7fdb18> with kwargs {'course_key': CourseLocator('edX', 'DemoX', 'Demo_Course', None, None)}. Response was: None
2018-02-17 00:07:56,176 INFO 20 [xmodule.modulestore.django] django.py:207 - Sent course_published signal to <function _listen_for_course_publish at 0x7f9c4c7fd8c0> with kwargs {'course_key': CourseLocator('edX', 'DemoX', 'Demo_Course', None, None)}. Response was: None
2018-02-17 00:07:56,188 WARNING 20 [xmodule.modulestore.xml] xml.py:233 - i4x://edX/DemoX/problem/Sample_ChemFormula_Problem has more than one definition
2018-02-17 00:07:56,189 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:56,194 WARNING 20 [xmodule.modulestore.xml] xml.py:233 - i4x://edX/DemoX/discussion/4d672c5893cb4f1dad0de67d2008522e has more than one definition
2018-02-17 00:07:56,299 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:56,314 WARNING 20 [xmodule.modulestore.xml] xml.py:233 - i4x://edX/DemoX/html/d45779ad3d024a40a09ad8cc317c0970 has more than one definition
2018-02-17 00:07:56,321 WARNING 20 [xmodule.modulestore.xml] xml.py:233 - i4x://edX/DemoX/html/55cbc99f262443d886a25cf84594eafb has more than one definition
2018-02-17 00:07:56,322 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:56,328 WARNING 20 [xmodule.modulestore.xml] xml.py:233 - i4x://edX/DemoX/discussion/ade92343df3d4953a40ab3adc8805390 has more than one definition
2018-02-17 00:07:56,336 WARNING 20 [xmodule.modulestore.xml] xml.py:233 - i4x://edX/DemoX/html/ed5dccf14ae94353961f46fa07217491 has more than one definition
2018-02-17 00:07:56,339 ERROR 20 [xmodule.vertical_block] vertical_block.py:131 - Unable to load child when parsing Vertical. Continuing...
Traceback (most recent call last):
  File "/openedx/edx-platform/common/lib/xmodule/xmodule/vertical_block.py", line 128, in definition_from_xml
    child_block = system.process_xml(etree.tostring(child, encoding='unicode'))
  File "/openedx/edx-platform/common/lib/xmodule/xmodule/modulestore/xml.py", line 196, in process_xml
    id_manager,
  File "/openedx/edx-platform/common/lib/xmodule/xmodule/x_module.py", line 1601, in xblock_from_node
    self._convert_reference_fields_to_keys(block)  # difference from XBlock.runtime
  File "/openedx/edx-platform/common/lib/xmodule/xmodule/x_module.py", line 1645, in _convert_reference_fields_to_keys
    field_value = getattr(xblock, field.name)
  File "/usr/local/lib/python2.7/dist-packages/xblock/fields.py", line 513, in __get__
    value = self.from_json(field_data.get(xblock, self.name))
  File "/usr/local/lib/python2.7/dist-packages/xblock/fields.py", line 819, in from_json
    raise TypeError('Value stored in a List must be None or a list, found %s' % type(value))
TypeError: Value stored in a List must be None or a list, found <type 'str'>
2018-02-17 00:07:56,341 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:56,346 WARNING 20 [xmodule.modulestore.xml] xml.py:233 - i4x://edX/DemoX/discussion/9f9e1373cc8243b985c8750cc8acec7d has more than one definition
2018-02-17 00:07:56,355 WARNING 20 [xmodule.modulestore.xml] xml.py:233 - i4x://edX/DemoX/html/030e35c4756a4ddc8d40b95fbbfff4d4 has more than one definition
2018-02-17 00:07:56,360 WARNING 20 [xmodule.modulestore.xml] xml.py:233 - i4x://edX/DemoX/video/0b9e39477cf34507a7a48f74be381fdd has more than one definition
2018-02-17 00:07:56,371 WARNING 20 [xmodule.modulestore.xml] xml.py:233 - i4x://edX/DemoX/html/78e3719e864e45f3bee938461f3c3de6 has more than one definition
2018-02-17 00:07:56,377 WARNING 20 [xmodule.modulestore.xml] xml.py:233 - i4x://edX/DemoX/problem/700x_proteinmake has more than one definition
2018-02-17 00:07:56,379 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:56,384 WARNING 20 [xmodule.modulestore.xml] xml.py:233 - i4x://edX/DemoX/discussion/ed01bcd164e64038a78964a16eac3edc has more than one definition
2018-02-17 00:07:56,397 WARNING 20 [xmodule.modulestore.xml] xml.py:169 - Non-unique url_name in xml.  This may break state tracking for content.  url_name=Video_Demonstrations.  Content=<vertical display_name="Video Demonstrations" parent_sequential_url="i4x://edX/DemoX/sequential/19a3
2018-02-17 00:07:56,433 WARNING 20 [xmodule.modulestore.xml] xml.py:233 - i4x://edX/DemoX/html/Lab_5B_Mosfet_Amplifier_Experiment has more than one definition
2018-02-17 00:07:56,434 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:56,439 WARNING 20 [xmodule.modulestore.xml] xml.py:233 - i4x://edX/DemoX/discussion/03f051f9a8814881a3783d2511613aa6 has more than one definition
2018-02-17 00:07:56,443 WARNING 20 [xmodule.modulestore.xml] xml.py:169 - Non-unique url_name in xml.  This may break state tracking for content.  url_name=New_Unit.  Content=<vertical display_name="New Unit" parent_sequential_url="i4x://edX/DemoX/sequential/simulations" ind
2018-02-17 00:07:56,474 WARNING 20 [xmodule.modulestore.xml] xml.py:233 - i4x://edX/DemoX/html/2b94658d2eee4d85ae13f83bc24cfca9 has more than one definition
2018-02-17 00:07:56,479 WARNING 20 [xmodule.modulestore.xml] xml.py:233 - i4x://edX/DemoX/problem/700x_editmolB has more than one definition
2018-02-17 00:07:56,481 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:56,485 WARNING 20 [xmodule.modulestore.xml] xml.py:233 - i4x://edX/DemoX/discussion/0aa7a3bdbe18427795b0c1a1d7c3cb9a has more than one definition
2018-02-17 00:07:56,498 WARNING 20 [xmodule.modulestore.xml] xml.py:233 - i4x://edX/DemoX/openassessment/b24c33ea35954c7889e1d2944d3fe397 has more than one definition
2018-02-17 00:07:56,499 WARNING 20 [root] mixins.py:522 - XBlock <class 'xblock.internal.DiscussionXBlockWithMixins'> does not contain field url_name
2018-02-17 00:07:56,505 WARNING 20 [xmodule.modulestore.xml] xml.py:233 - i4x://edX/DemoX/discussion/12ad4f3ff4c14114a6e629b00e000976 has more than one definition
2018-02-17 00:07:57,095 INFO 20 [openedx.core.djangoapps.content.course_overviews.models] models.py:147 - Updating course overview for course-v1:edX+DemoX+Demo_Course.
2018-02-17 00:07:57,132 INFO 20 [xmodule.modulestore.django] django.py:207 - Sent course_published signal to <function listen_for_course_publish at 0x7f9c67a3f410> with kwargs {'course_key': CourseLocator('edX', 'DemoX', 'Demo_Course', None, None)}. Response was: None
2018-02-17 00:07:57,134 INFO 20 [xmodule.modulestore.django] django.py:207 - Sent course_published signal to <function listen_for_course_publish at 0x7f9c4c9adaa0> with kwargs {'course_key': CourseLocator('edX', 'DemoX', 'Demo_Course', None, None)}. Response was: None
2018-02-17 00:07:57,135 INFO 20 [xmodule.modulestore.django] django.py:207 - Sent course_published signal to <function _update_block_structure_on_course_publish at 0x7f9c4c9b36e0> with kwargs {'course_key': CourseLocator('edX', 'DemoX', 'Demo_Course', None, None)}. Response was: None
2018-02-17 00:07:57,136 INFO 20 [xmodule.modulestore.django] django.py:207 - Sent course_published signal to <function _listen_for_course_publish at 0x7f9c4c7fd578> with kwargs {'course_key': CourseLocator('edX', 'DemoX', 'Demo_Course', None, None)}. Response was: None
2018-02-17 00:07:57,138 INFO 20 [xmodule.modulestore.django] django.py:207 - Sent course_published signal to <function _listen_for_course_publish at 0x7f9c4c7fda28> with kwargs {'course_key': CourseLocator('edX', 'DemoX', 'Demo_Course', None, None)}. Response was: None
2018-02-17 00:07:57,139 INFO 20 [xmodule.modulestore.django] django.py:207 - Sent course_published signal to <function trigger_update_xblocks_cache_task at 0x7f9c4c7fdb18> with kwargs {'course_key': CourseLocator('edX', 'DemoX', 'Demo_Course', None, None)}. Response was: None
2018-02-17 00:07:57,141 INFO 20 [xmodule.modulestore.django] django.py:207 - Sent course_published signal to <function _listen_for_course_publish at 0x7f9c4c7fd8c0> with kwargs {'course_key': CourseLocator('edX', 'DemoX', 'Demo_Course', None, None)}. Response was: None
Seeding forum roles for course course-v1:edX+DemoX+Demo_Course
```

**Comment by regisb** [comment]
Ok so there are multiple things happening here. First, the errors logged during import are probably caused by an outdated course structure (the [demo course](https://github.com/edx/edx-demo-course) was not updated for a long time).

Secondly, the course "about" page could not be viewed because the CMS was pointing to the wrong mongodb database. (see https://github.com/regisb/openedx-docker/commit/189ea3e7dc9695f7287f85202f57746728b42f5c) To fix this, pull the latest version of the repo and then run: 

    ./configure --silent
    make build

Finally, if you still cannot view the demo course "about" page after re-importing, make sure that you are logged-in in the LMS.

**Comment by regisb** [comment]
@natea feel free to re-open if the proposed solution does not solve your problem.

**Comment by regisb** [comment]
@natea thanks to a tip by @dannielarriola I managed to fix this issue! See #11 for details.
